src/Form/FullFileDeletionForm.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Drupal\tide_media\Form;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Entity\ContentEntityConfirmFormBase;
+use Drupal\Core\Entity\EntityRepositoryInterface;
+use Drupal\Core\Entity\EntityTypeBundleInfoInterface;
+use Drupal\Core\File\Exception\FileNotExistsException;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\file\Entity\File;
+use Drupal\file\FileUsage\FileUsageInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a form for deleting a media entity.
+ */
+abstract class FullFileDeletionForm extends ContentEntityConfirmFormBase {
+
+  /**
+   * The media being deleted.
+   *
+   * @var \Drupal\file\FileInterface
+   */
+  protected $entity;
+
+  /**
+   * The File Usage Service.
+   *
+   * @var \Drupal\file\FileUsage\FileUsageInterface
+   */
+  protected $fileUsage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(FileUsageInterface $file_usage, EntityRepositoryInterface $entity_repository, EntityTypeBundleInfoInterface $entity_type_bundle_info = NULL, TimeInterface $time = NULL) {
+    $this->fileUsage = $file_usage;
+
+    parent::__construct($entity_repository, $entity_type_bundle_info, $time);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('file.usage'), $container->get('entity.repository'), $container->get('entity_type.bundle.info'), $container->get('datetime.time'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $parent = parent::buildForm($form, $form_state);
+    $file = $this->getCurrentFile();
+    // If no file attached, just return.
+    if (!$file) {
+      return $parent;
+    }
+    // Basic settings.
+    $parent['info'] = [
+      '#type' => 'table',
+      '#header' => [
+        'file name',
+        'url',
+        'delete',
+      ],
+    ];
+    /** @var \Drupal\file\FileStorageInterface $file_storage */
+    $file_storage = \Drupal::entityTypeManager()->getStorage('file');
+    // Try to get a search keyword.
+    $key = preg_replace('/_[^_.]*\./', '.', $file->getFilename());
```

Comment:
@vincent-gao what will happen if those _0, _1, etc. files are referenced by other media entities?

Reply:
the files will be deleted, but media entities will remain. 
![image](https://user-images.githubusercontent.com/8788145/96396871-ee805b00-1213-11eb-921b-bc4439760d31.png)
I will update the table where it shows which media the file belongs to.  users must select which files they want to delete.
cc @anthony-malkoun

Reply:
hi @vincent-gao @anthony-malkoun,

From your screenshot, I could see that the Full Deletion lists more files than required and it may cause an accidental deletion. I think we only need to list files with incremental suffix (`<BASENAME>_<DIGITS>.<EXTENTION>`) instead of a similar pattern. 

If the file being deleted is `/sites/default/files/sample.pdf`, we should only list `sample_0.pdf`, `sample_1.pdf`, etc. in the same directory `/sites/default/files`, but not `abc-sample.pdf`, `sample.rtf`, nor `/sites/default/files/2020-10/sample.pdf`.

Reply:
Thanks @sonnykt . good suggestion.

Reply:
@vincent-gao it may be useful if we also show the timestamp and file status in the list

Reply:
@sonnykt agreed!

Reply:
![image](https://user-images.githubusercontent.com/8788145/96453258-eea94680-1265-11eb-8a52-20266ee1811a.png)

Reply:
hi @vincent-gao @anthony-malkoun,

jm2c, should we keep the bulk deletion UI aligned with other Views?